### DISCUSSION TITLE
Possibility of Bundling `static-curl`

### DISCUSSION THREAD

**Discussion by mohammed90** [Orange-OpenSource/hurl#3225] [Ideas]
I noticed Hurl uses the system's curl/libcurl, virtue of the `curl-sys` crate. Sometimes I use GitHub Codespaces for away-from-home development, and they only offer Ubuntu 20.04, which only has (lib)curl/7.68. The latest `curl-sys` supports curl 8.0+, and it also offers `static-curl` to not rely on the system-provided (lib)curl. Using `static-curl` allows having the latest fixes of curl. 

This is a request due to niche environment, and it isn't a dire need. This whole thing was visible to me because I encountered a curl bug that had to do with GOAWAY and was fixed somewhere after 7.68.

**Comment by jcamiel** [comment]
Hi @mohammed90 

I think we can look at exposing the flag `static-curl` with a caveat: this will only be available for user willing to recompile themself the binary `hurl`. Unless there is a very strong need, it will be difficult for us to support/provide "static curl" Hurl binaries and "dynamic curl" binaries. 

BTW, we have ourself encountered issues with GitHub Action runners / Ubuntu version / curl 7.68. Will it be a possibility for you to upgrade `curl` in your Ubuntu 20.04 env?

**Reply by mohammed90** [reply]
> Will it be a possibility for you to upgrade `curl` in your Ubuntu 20.04 env?

It's whatever GitHub Codespaces gives. I managed to replace `/usr/bin/curl` with latest curl, but `libcurl` is tricky for me to replace. I think I have to use `LD_PRELOAD*` env var or something, but I haven'g figured it out yet.

The need is not strong. It adds convenience for me and allows me to work on stuff while browsing GitHub from the browser of another machine.

**Reply by jcamiel** [reply]
@mohammed90 noted. I will try to set up a build with "static" libcurl to see what's are the consequences for a start!
